airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/MemoryAwareMessageBatch.java

```diff
@@ -64,9 +63,9 @@ public void close() throws Exception {
    *
    * @return list of states that can be flushed
    */
-  public List<PartialStateWithDestinationStats> flushStates(final Map<Long, Long> stateIdToCount) {
+  public void flushStates(final Map<Long, Long> stateIdToCount) {
     stateIdToCount.forEach(stateManager::decrement);
-    return stateManager.flushStates();
+    stateManager.flushStates();
```

Comment:
Curious, does the locking break if we return the ids to flush and flush them in the FlushWorkers instead?

That way, all the actual action happens in the flush workers, and not in objects scattered around. Mainly a cleanliness suggestion.